### ISSUE TITLE
[explicit-member-accessibility] false positive on pbulic constructor

### ISSUE THREAD

**Issue by SleeplessByte** [typescript-eslint/typescript-eslint#362]
**Repro**

```JSON
{
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "project": "./tsconfig.json"
  },
  "plugins": ["@typescript-eslint"],
  "extends": [
    "plugin:@typescript-eslint/recommended",
    "plugin:react/recommended",
    "prettier",
    "prettier/@typescript-eslint",
    "prettier/babel",
    "prettier/react"
  ],
  "rules": {
    "@typescript-eslint/no-explicit-any": "off",
    "@typescript-eslint/no-parameter-properties": "off",
    "@typescript-eslint/no-unused-vars": ["on", { "argsIgnorePattern": "^_", "varsIgnorePattern": "^ignored?$" }]
  },
}
```

```TS
export class SourceNotOk extends Error {

  public constructor(public readonly error: any, public readonly source: string) {
    super(`Unable to load ${source}: ${error}`)
    this.error = error

    Error.captureStackTrace(this, this.constructor)
  }
}
```

**Expected Result**

All good

**Actual Result**

```
constructor SourceNotOk(error: any, source: string): SourceNotOk

Missing accessibility modifier on method definition constructor.eslint(@typescript-eslint/explicit-member-accessibility)
```
**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.4.2` |
| `@typescript-eslint/parser`        | `1.4.2` |
| `TypeScript`                       | `3.3.3333` |
| `ESLint`                           | `5.15.2` |
| `node`                             | `10.7.0` |
| `yarn`                              | `1.12.1` |

**Comment by SleeplessByte** [comment]
Resolved itself after restarting twice.
